arangod/Statistics/StatisticsFeature.cpp

```diff
@@ -186,19 +194,27 @@ void StatisticsFeature::start() {
   }
 
   _statisticsThread.reset(new StatisticsThread);
-  _statisticsWorker.reset(new StatisticsWorker(*vocbase));
 
   if (!_statisticsThread->start()) {
     LOG_TOPIC(FATAL, arangodb::Logger::STATISTICS)
         << "could not start statistics thread";
     FATAL_ERROR_EXIT();
   }
 
-  if (!_statisticsWorker->start()) {
-    LOG_TOPIC(FATAL, arangodb::Logger::STATISTICS)
+  // force history disable on Agents
+  if (arangodb::ServerState::instance()->isAgent()) {
```

Comment:
IMHO users should be able to enable the statistics history when they set the option.
So we should only turn it off when the user has *not* set the option using startup parameters.

Reply:
Max and Kaveh were greatly concerned about the impact of any statistics with agent.  That is why the statistics module was previously disabled for the agent.  This was a compromise.  Since agent statistics have never been available at all, this exception will not change anything the user previously expected.

Reply:
True. But honoring the option when set will also not worsen anything. And will be downwards-compatible.